uPortal-webapp/src/main/webapp/WEB-INF/jsp/Search/searchRest.jsp

```diff
@@ -0,0 +1,212 @@
+    for additional information regarding copyright ownership.
+    Apereo licenses this file to you under the Apache License,
+    Version 2.0 (the "License"); you may not use this file
+    except in compliance with the License.  You may obtain a
+    copy of the License at the following location:
+      http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+--%>
+<%@ include file="/WEB-INF/jsp/include.jsp"%>
+
+<template id="search-results-tab-header-template">
+    <a class="mdl-tabs__tab"></a>
+</template>
+
+<template id="search-results-tab-panel-template">
+    <div class="mdl-tabs__panel">
+        <ul class="mdl-list"></ul>
+    </div>
+</template>
+
+<template id="search-result-item-template">
+    <li class="mdl-list__item mdl-list__item--three-line">
+        <span class="mdl-list__item-primary-content">
+            <i class="material-icons mdl-list__item-avatar"></i>
+            <span class="up-list-item-title"></span>
+            <span class="mdl-list__item-text-body">
+                <dl></dl>
+            </span>
+        </span>
+        <span class="mdl-list__item-secondary-content">
+            <a class="mdl-list__item-secondary-action" href="#">
+                <i class="material-icons">
+                    open_in_browser
+                </i>
+            </a>
+        </span>
+    </li>
+</template>
+
+<template id="search-result-item-detail-template">
+    <dt></dt>
+    <dd></dd>
+</template>
+
+<template id="search-result-no-results-template">
+    <div class="h3 text-center">
+         <i class="material-icons">warning</i>
+         No Results Found
+         <i class="material-icons">warning</i>
+     </div>
+</template>
+
+<!-- TODO: use existing material design lite from resource server -->
+<script src="https://code.getmdl.io/1.3.0/material.min.js"></script>
+<link rel="stylesheet" href="https://code.getmdl.io/1.3.0/material.indigo-pink.min.css">
+<!-- above 2 are required for Material Design -->
+
+<!-- TODO: move material design icons, lodash/lodash, github/fetch, webcomponents/template, and taylorhakes/promise-polyfill to resource server -->
+<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
+<!-- similar to fontawesome? -->
+<script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.4/lodash.min.js"></script>
+<!-- Does the above replace underscore? -->
+<script src="https://rawgit.com/github/fetch/v2.0.3/fetch.js"></script>
+<script src="https://rawgit.com/webcomponents/template/v1.0.0/template.js"></script>
+<script src="https://rawgit.com/taylorhakes/promise-polyfill/6.0.2/promise.js"></script>
```

Comment:
We already plan to remove these imports (based on previous discussion), but I want to point out that they change the appearance of tabs, portlet chrome, and other things outside the content produced by this jsp.

Reply:
We have no intentions of removing the imports lines 69-73.
Those are polyfills to support JavaScript API's missing from Internet Explorer.
Ideally the will be removed, but not until IE support is dropped.

Material Design Lite lines 62 and 63 can impact surround DOM as is,
but when this moves to leverage the scoped version provided by the resource server, that version has not no external impact.

Reply:
https://github.com/Jasig/resource-server/pull/67, https://github.com/Jasig/resource-server/pull/69, https://github.com/Jasig/resource-server/pull/70 have been opened to add polyfills

Reply:
I wish we could sit down in the same place and hash this item out completely.

I'm not familiar with the polyfills.  I do see a lot of non-jQuery JavaScript in the JSP that (afaik) jQuery handles at least as well.  jQuery is already available in the skin/portal, requiring no additional imports.

I see a lot of method calls directly on DOM elements that don't look familiar.  I could have sworn that -- when I started to work with DOM manipulation circa 2007 -- there was no such thing as (for example) `document.querySelector`.  I would have found that kind of thing very useful.  I'm guessing that's what these polyfills are about.

Bootstrap is also already a part of the skin/portal.  Using it requires no imports, and is likely to produce a result that "blends in well" with the rest of the portal UI.

You mention a "scoped version" of Material Design Lite -- is that something we engineered?  (We did something like that for Bootstrap originally;  I don't feel like it went spectacularly.)

Reply:
> I do see a lot of non-jQuery JavaScript in the JSP that (afaik) jQuery handles at least as well. jQuery is already available in the skin/portal, requiring no additional imports.

jQuery methods have been replaced with their native browser equivalents.
In Firefox, Chrome, Edge, and Safari this code works fine without any polyfills.
The polyfills are needed to support Internet Explorer, which is still on the official support list.

The items that are being polyfillled
1. the HTML `<template>` tag (https://caniuse.com/#feat=template)
2. the JavaScript `Promise` API (`.then` and `.catch`) (https://caniuse.com/#feat=promises)
3. the JavaScript `fetch` API, the successor to `XMLHttpRequest` (https://caniuse.com/#feat=fetch)

Yes, jQuery *could* be used to do the same things.
But using native web standards makes more sense.

>  could have sworn that -- when I started to work with DOM manipulation circa 2007 -- there was no such thing as (for example) document.querySelector. I would have found that kind of thing very useful. I'm guessing that's what these polyfills are about.

`document.querySelector` was introduced in 2009, so a few years later. 
ref: https://caniuse.com/#feat=queryselector
ref: https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector
It has wide enough IE support that no polyfill was added.

> You mention a "scoped version" of Material Design Lite -- is that something we engineered?

That is something we engineered.
https://github.com/Jasig/resource-server/pull/59

Reply:
@ChristianMurphy is the `template.js` available in the resource-server?

Reply:
https://github.com/Jasig/resource-server/pull/73

Reply:
All `.js` files used in this JSP are now available in the `resource-server` and being pulled from there.